lib/ansible/modules/tacp/tacp_instance.py

```diff
@@ -80,278 +78,713 @@
+                                             MODULE.params['portal_url'])
+API_CLIENT = tacp.ApiClient(CONFIGURATION)
+
+RESOURCES = {
+    'app': tacp_utils.ApplicationResource(API_CLIENT),
+    'app_group': tacp_utils.ApplicationGroupResource(API_CLIENT),
+    'datacenter': tacp_utils.DatacenterResource(API_CLIENT),
+    'migration_zone': tacp_utils.MigrationZoneResource(API_CLIENT),
+    'storage_pool': tacp_utils.StoragePoolResource(API_CLIENT),
+    'template': tacp_utils.TemplateResource(API_CLIENT),
+    'update_app': tacp_utils.UpdateApplicationResource(API_CLIENT),
+    'vlan': tacp_utils.VlanResource(API_CLIENT),
+    'vnet': tacp_utils.VnetResource(API_CLIENT)
+}
+
+
+def fail_with_reason(reason):
+    RESULT['msg'] = reason
+    MODULE.fail_json(**RESULT)
+
+
+def fail_and_rollback_instance_creation(reason, instance):
+    RESULT['msg'] = reason + "\n Rolled back application instance creation."
+    RESOURCES['app'].delete(instance.uuid)
+    MODULE.fail_json()
+
+
+def get_parameters_to_create_new_application(playbook_instance):
+    """Given the input configuration for an instance, generate
+        parameters for creating the appropriate payload elsewhere.
 
+    Args:
+        playbook_instance (dict): The specified instance configuration from
+        the playbook input
+
+    Returns:
+        dict: The parameters to be provided to an ApiCreateApplicationPayload
+            object.
+    """
+    parameters_to_create_new_application = {}
+    parameters_to_create_new_application['instance_name'] = playbook_instance['name']  # noqa
+
+    resources = [
+        'datacenter',
+        'migration_zone',
+        'storage_pool',
+        'template'
+    ]
+
+    for resource in resources:
+        resource_list = RESOURCES[resource].filter(
+            name=playbook_instance[resource])
+        if not resource_list:
+            fail_with_reason("{} resource {} does not exist, cannot continue.".format(  # noqa
+                resource, playbook_instance[resource]
+            ))
+        resource_uuid = resource_list[0].uuid
+
+        parameters_to_create_new_application['{}_uuid'.format(
+            resource)] = resource_uuid
+
+        if resource == 'template':
+            template_boot_order = resource_list[0].boot_order
+            parameters_to_create_new_application['boot_order'] = template_boot_order  # noqa
+
+    if playbook_instance['application_group']:
+        uuid = RESOURCES['app_group'].get_uuid_by_name(
+            playbook_instance['application_group']
+        )
+        if uuid is None:
+            resp = RESOURCES['app_group'].create(playbook_instance['application_group'],  # noqa
+                        parameters_to_create_new_application['datacenter_uuid'])  # noqa
+            uuid = resp.object_uuid
+
+        parameters_to_create_new_application['application_group_uuid'] = uuid
+
+    network_payloads = []
+    vnic_payloads = []
+    template_vnics = [boot_device for boot_device
```

Comment:
It would be nicer to break the line after the sequence 
```suggestion
    template_vnics = [boot_device for boot_device in template_boot_order
                      if boot_device.vnic_uuid]
```